client/src/common/client.ts

```diff
@@ -750,6 +766,13 @@ export abstract class BaseLanguageClient implements FeatureClient<Middleware, La
 				exclude = (params as DidOpenTextDocumentParams)?.textDocument.uri;
 			}
 			await this.sendPendingFullTextDocumentChanges(connection, exclude);
+
+			const generalMiddleware = this._clientOptions.middleware?.provideRequestOrNotification;
+
+			if (generalMiddleware) {
```

Comment:
I started to make checks explicit (e.g. generalMiddleware !== undefined) since it is easier to read.